pkg/registry/apps/statefulset/strategy.go

```diff
@@ -103,7 +106,31 @@ func (statefulSetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 	if !apiequality.Semantic.DeepEqual(oldStatefulSet.Spec, newStatefulSet.Spec) {
 		newStatefulSet.Generation = oldStatefulSet.Generation + 1
 	}
+}
 
+// dropStatefulSetDisabledFields drops fields that are not used if their associated feature gates
+// are not enabled.
+// The typical pattern is:
+//     if !utilfeature.DefaultFeatureGate.Enabled(features.MyFeature) && !myFeatureInUse(oldSvc) {
+//         newSvc.Spec.MyFeature = nil
+//     }
+func dropStatefulSetDisabledFields(newSS *apps.StatefulSet, oldSS *apps.StatefulSet) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetMinReadySeconds) {
+		if !minReadySecondsFieldsInUse(oldSS) {
+			newSS.Spec.MinReadySeconds = int32(0)
+		}
+	}
+}
+
+// minReadySecondsFieldsInUse returns true if fields related to StatefulSet minReadySeconds are set and
+// are greater than 0
+func minReadySecondsFieldsInUse(ss *apps.StatefulSet) bool {
+	if ss == nil {
+		return false
+	} else if ss.Spec.MinReadySeconds >= 0 {
```

Comment:
This is ok.  As long as omit empty is present in the serialization (it is) then this is allowed.